src/index.test.ts

```diff
@@ -63,6 +64,58 @@ describe("voices", () => {
 });
 
 describe("tts.websocket", () => {
+  let phonicWebSocket: PhonicWebSocket;
+  let allMessages: PhonicWebSocketResponseMessage[] = [];
+
+  const waitForMessages = async (
+    count: number,
+    timeout = 20_000,
+  ): Promise<PhonicWebSocketResponseMessage[]> => {
+    const startTime = Date.now();
+
+    // Wait for expected count or timeout
+    while (allMessages.length < count) {
+      if (Date.now() - startTime > timeout) {
+        throw new Error(`Timeout waiting for ${count} messages`);
+      }
+      await Bun.sleep(100);
+    }
```

Comment:
This while loop with the arbitrary 100ms sleep doesn't feel right.
My understanding is that we want to say "wait until all messages are received" and have some max idle time defined that we assume that if this time passed since the last message then there will be no more messages.
What do you think about something like this?
```typescript
describe("tts.websocket", () => {
  let allMessagesReceived = null;
  const maxIdleTime = 5000; // If we don't receive messages for 5 seconds, we consider that we received all of them.

  beforeEach(async () => {
    ...
    const messages = [];
    let messagesTimeoutId = null;
    let allMessagesReceivedResolve = null;

    phonicWebSocket.onOpen(() => {
      allMessagesReceived = new Promise(resolve => {
        allMessagesReceivedResolve = resolve;
      });
    });

    phonicWebSocket.onMessage((message) => {
      messages.push(message);
  
      if (messagesTimeoutId !== null) {
        clearTimeout(messagesTimeoutId)
      }
  
      messagesTimeoutId = setTimeout(() => {
        allMessagesReceivedResolve(messages);
      }, maxIdleTime);
    });
  });
  
  test("...", async () => {
    phonicWebSocket.generate({ text: "Hello world" });
    phonicWebSocket.flush();
    
    const messages = await allMessagesReceived;

    // assert `messages` here
  });
});
```

Reply:
Done, let me know what you think. I had to change your implementation slightly to support awaiting on the promise multiple times.